### ISSUE TITLE
problems when running fairSIM in my computer

### ISSUE THREAD

**Issue by lollows** [fairSIM/fairSIM#33]
Hi，
   I am using eclipse and jdk(version 11.0.1) to run fairSIM(the develop branch) in my computer(windows 7 64). After a fresh clone from git, the ImageJ jar file(ij-1.52i.jar) has to be added to ./external. Why can i find the fairSIM plugin in the fuji. However, I can find it in the release version(fairSIM-1.4.1). And I had a successful run in my computer using other git projects(imagej1-plugins-ide-setup). Any suggestions are helpful. Thanks in advance.
lollows

**Comment by mueller-physics** [comment]
Hi lollows,

I think there might be some issues with external dependencies. To build fairSIM, you need as external libraries:

* ImageJ (ij...jar) in version 1.47v or later 
* The JTransforms fairSIM fork from [here](https://github.com/fairSIM/JTransforms/releases)
* The original JTransforms from [here](https://github.com/wendykierp/JTransforms)
* Depending on the JTransforms version, maybe JLargeArrays from [here](https://gitlab.com/ICM-VisLab/JLargeArrays)
* Apache Common Math version 3.6.1 or later

I don't use Eclipse to build fairSIM, so I don't know how that handles library dependencies. In general, you can:
* copy the files by hand as you did
* use the script in external/get-dependencies.sh (though on Windows you'd need some extra software)
* use the [Maven](https://maven.apache.org/) build system

Some of my colleagues who develop on Netbeans / Windows use either the manual option or the automated maven builds. I develop on Linux (and Windows using [msys2](https://www.msys2.org/)), where I use the script above and the make file included in the project.

If you want the resulting plugin file to run on arbitrary ImageJ / Fiji installations, you also have to ensure to compile it with Java version 1.6 compatibility. Both the make file and the maven build do so automatically.

I hope this helps.

**Comment by lollows** [comment]
Hi,
Thank you for your immediate response. I  have changed my java version(jdk 1.6.0_45u) and used the Maven build system to run fairSIM again. Everything seems to be ok except that I cann't find the fairSIM plugin in the fuji. Is there other steps (for example, installing the fairSIM plugin in fiji) to do the work?
![my result](https://user-images.githubusercontent.com/31273965/50197826-d6272700-0383-11e9-9086-33ad578fbf57.jpg)
lollows

**Comment by mueller-physics** [comment]
Yes, for fairSIM to work as an ImageJ / FIJI plugin (the most common use case),  the final compiled jar file (found in the "target" directory when using maven) needs to be copied into the "plugin" folder of the FIJI installation. The jar file should have a "_" in its name, and there must not be duplicate versions of the plugin in the folder at the same time.

If you use the maven build system, you also need to provide the JTransforms files to FIJI, by either copying them into into its "jars" folder. The standard fairSIM release gets around this by bundling both fairSIM and the forked JTransforms into a single, self-contained file.

If you want to develop on the code, copying the plugin each time becomes cumbersome. In that case, it might be easier to run fairSIM as the main program, and use ImageJ in "embedded mode" for input / output only. 
See the main methods of "org/fairsim/tests/TestPlugin.java" or "org/fairsim/fiji/TestRLdeconvSIM.java" for examples of this

**Comment by lollows** [comment]
Hi,
    Thanks for your kind helps. Now, I can run fairSIM in eclipse or netbeans perfectly.However, sonatype no longer supports TLSv1.1 and below (effective, June 18th, 2018). So, if you use maven to build the project with JDK 1.6 (not JDK 1.8 or high version),there will be some problems such as "Could not transfer artifact https://repo.maven.apache.org/maven2): Received fatal alert: protocol_version -> [Help 1]". As a result ,you must modify the pom.xml.  In my opinion, it's a little bit difficlut for the  beginner. Also, the netbeans project and eclipse have some differents in building the projects with maven. Maybe later(during my free time), I'll write some something in details about running the fairSIM in eclipse&windows .
lollows

**Comment by lollows** [comment]
For the problem above,there are some good solutions to refering:(1)https://stackoverflow.com/questions/50946420/could-not-transfer-artifact-https-repo-maven-apache-org-maven2-received-fat/50956622
(2)https://blog.csdn.net/u010800201/article/details/81285620

**Comment by mueller-physics** [comment]
Sorry for the delay, new years holidays...
Thanks for the heads up: On my Linux system (Ubuntu 18.04) the maven build runs error-free, but I haven't tested it on Windows... I'll try to reproduce & fix this once I have a build system set up on one of our Windows computers.
